A72-open-telemetry-tracing.md

```diff
@@ -105,94 +133,181 @@ class OpenTelemetryPluginBuilder {
+  name resolution process.
+* When the call is closed, set RPC status and end the parent span. gRPC status "OK" 
+  is recorded with status "OK", while other gRPC statuses are marked as "ERROR". 
+  Non-"OK" statuses include their code as a description, the same below.
 
 On attempt span:
-* When span is created, add attribute `previous-rpc-attempts` that captures the
-  number of preceding attempts for the RPC, and attribute `transparent-retry` that
-  shows whether stream is a transparent retry.
-* When the stream is created on transport, annotate delayed load balancer pick
-  complete, if any.
-* When an outbound message has been sent, add message events to capture seq no.,
-  type(SENT), uncompressed message size, and compressed message size if any
-  compression. The seq no. is a sequence of integer numbers starting from 0
-  to identify sent messages within the stream. The size is the total attempt message
-  bytes without encryption, not including grpc or transport framing bytes.
-* When an inbound message has been received from the transport, add message
-  events to capture seq no., type(Received), wire message size, and uncompressed
-  message size if any decompression. The seq no. is a sequence of integer numbers
-  starting from 0 to identify received messages within the stream.
+* When span is created, set the attribute with key `previous-rpc-attempts` and an 
+  integer value representing the count of previous attempts made for the RPC.
+* When span is created, set the attribute with key `transparent-retry` and a 
+  boolean value indicating whether the stream is undergoing a transparent retry.
+* If the RPC experienced load balancer pick delay, add an Event with the name 
+  "Delayed LB pick complete" upon creation of the stream on the transport. 
+* When the application sends an outbound message to the transport, add an Event 
+  with name "Outbound message sent" and the following attributes:
+  * key `message.event.type` with string value "SENT".
+  * key `message.message.id` with integer value of the seq no. The seq no. is a sequence
+    of integer numbers starting from 0 to identify sent messages within the stream, the same below.
+  * key `message.event.size.uncompressed` with integer value of uncompressed 
+    message size. The size is the total attempt message bytes without encryption,
+    not including grpc or transport framing bytes, the same below.
+  * If any compression, key `message.event.size.compressed` with integer value
+    of compressed message size.
+* When an inbound message has been received from the transport, add an Event
+  with name "Inbound message read" and the following attributes:
+  * key `message.event.type` with String value "RECEIVED".
+  * key `message.message.id` with integer value of the seq no.
+  * key `message.event.size.compressed` with integer value of wire message size. 
+  * If any compression, key `message.event.size.uncompressed` with integer value
```

Comment:
Isn't this backwards?  If there is no compression, I'd expect us to report the uncompressed size, not the compressed size.

Same for server side.

Reply:
when received from wire, we don't know there is compression or not, so we call it in general "message.event.size.compressed", indicating the wire size. This is similar to OpenCensus tracing info in java.

Reply:
I think this approach results in somewhat inconsistant naming when compression is not used: we will wind up reporting only `message.event.size.compressed` for inbound messages but only `message.event.size.uncompressed` for outbound messages.

I can think of a couple possible alternatives here (other suggestions welcome):
1. If compression is not used, report message size as just "message.event.size", without labelling it as either compressed or uncompressed.
2. Instead of calling these attributes "compressed" and "uncompressed", report them as "application" and "wire" sizes.  Always report the wire size, and report the application size only if compression is used.

Reply:
I think we should consider parity from opencensus first, and make possible changes when we see there is improvements to make. So report `message.event.size` without knowing it needs compression or decompression, and "compressed/uncompressed" for a second event sounds fair to me. This would be less of a problem for Java/Go that reports one event only.

Reply:
I think we should define a schema that is most clear for users rather than being bound by what OpenCensus did.

As currently written, if compression is enabled, then `message.event.size` will be the uncompressed size for outbound messages but the compressed size for inbound messages.  I think that's going to be very confusing.  I think we should define a schema that makes sense both when compression is enabled and disabled.

I would prefer one of the following options:
1. We use different names depending on whether or not compression is enabled:
   -  If compression is enabled, then we use `message.event.size.compressed` and `message.event.size.uncompressed`.
   - If compression is disabled, then we use just `message.event.size`.
2. Instead of calling these attributes "compressed" and "uncompressed", report them as "application" and "wire" sizes. Always report the wire size, and report the application size only if compression is used.

Reply:
I did option1. The "application"/"wire" sounds more literal but is less precise. I'm nervous they will introduce further discussions.